File: другое/тинькофф.аисд/1/I.ts
const ln = 4;
const arr = [1, 3, 4, 2];

const lst = new Array(ln).fill(0);
const ans: number[] = [1];
let cL = 0;
let zero = ln - 1;
for (const item of arr) {
  lst[item - 1] = 1;
  if (item - 1 === zero) {
    for (let i = zero; i >= 0; i--) {
      if (lst[i] === 0) {
        zero = i;
        break;
      } else {
        cL--;
      }
    }
  } else {
    cL++;
  }

  ans.push(cL + 1);
}
console.log(ans.join(" "));

File: другое/тинькофф.аисд/1/I.py
ln = int(input())
arr = list(map(int, input().split()))

lst = [0] * ln
ans = [1]
cL = 0
zero = ln - 1

for item in arr:
    lst[item - 1] = 1
    if item - 1 == zero:
        for i in range(zero - 1, -1, -1):
            if lst[i] == 0:
                zero = i
                break
            else:
                cL -= 1
    else:
        cL += 1

    ans.append(cL + 1)

print(" ".join(map(str, ans)))
